File: backend/api/serializers/loyalty.py
from rest_framework import serializers

from ..models import LoyaltyHistory, LoyaltySetting


class LoyaltyHistorySerializer(serializers.ModelSerializer):
    client_name = serializers.CharField(source='client.name', read_only=True)
    facture_numero = serializers.CharField(source='facture.numero_facture', read_only=True)
    created_by_name = serializers.CharField(source='created_by.username', read_only=True)
    type_display = serializers.CharField(source='get_type_transaction_display', read_only=True)

    class Meta:
        model = LoyaltyHistory
        fields = '__all__'
        read_only_fields = ['client', 'facture', 'points', 'solde_apres', 'montant', 'created_by', 'created_at']


class LoyaltySettingSerializer(serializers.ModelSerializer):
    class Meta:
        model = LoyaltySetting
        fields = '__all__'

File: frontend/frontend/src/types/loyalty.ts
export interface LoyaltyHistoryEntry {
    id: number;
    client: number;
    client_name: string;
    facture: number | null;
    facture_numero: string | null;
    type_transaction: 'GAIN' | 'UTILISATION' | 'REMISE_AUTO' | 'AJUSTEMENT';
    type_display: string;
    points: number;
    solde_apres: number;
    montant: string;
    created_by: number | null;
    created_by_name: string;
    created_at: string;
    notes: string;
}

export interface LoyaltySettings {
    id: number;
    amount_per_point: string;
    point_value: string;
    auto_reward_threshold: number;
    auto_reward_percent: string;
}
